examples/external-otlp-tonic-tokio/src/main.rs

```diff
@@ -0,0 +1,107 @@
+//! This should show how to connect to a third party collector like
+//! honeycomb or lightstep using tonic with tls and using tokio as reactor.
+//! To run this you have to specify a few environment variables like in the example:
+//! ```shell
+//! OTLP_TONIC_ENDPOINT=https://api.honeycomb.io:443 \
+//! OTLP_TONIC_X_HONEYCOMB_TEAM=token \
+//! OTLP_TONIC_X_HONEYCOMB_DATASET=dataset '
+//! cargo run --bin external-otlp-tonic-tokio
+use opentelemetry::trace::TraceError;
+use opentelemetry::{global, sdk::trace as sdktrace};
+use opentelemetry::{
+    trace::{TraceContextExt, Tracer},
+    Key, KeyValue,
+};
+use tonic::{
+    metadata::{MetadataKey, MetadataMap},
+    transport::ClientTlsConfig,
+};
+use url::Url;
+
+use std::{env::vars, str::FromStr, time::Duration};
+use std::{
+    env::{remove_var, var},
+    error::Error,
+};
+
+// Use the variables to try and export the example to any external collector that accepts otlp
+// like: oltp itself, honeycomb or lightstep
+const ENDPOINT: &str = "OTLP_TONIC_ENDPOINT";
+const HEADER_PREFIX: &str = "OTLP_TONIC_";
+
+fn init_tracer() -> Result<(sdktrace::Tracer, opentelemetry_otlp::Uninstall), TraceError> {
+    let endpoint = var(ENDPOINT).unwrap_or_else(|_| {
+        panic!(
+            "You must specify and endpoint to connect to with the variable {:?}.",
+            ENDPOINT
+        )
+    });
+    let endpoint = Url::parse(&endpoint).expect("endpoint is not a valid url");
+    remove_var(ENDPOINT);
+    let mut metadata = MetadataMap::new();
+    for (key, value) in vars()
+        .filter(|(name, _)| name.starts_with(HEADER_PREFIX))
+        .map(|(name, value)| {
+            let header_name = name
+                .strip_prefix(HEADER_PREFIX)
+                .unwrap()
+                .replace("_", "-")
+                .to_ascii_lowercase();
+            (header_name, value)
+        })
+    {
+        metadata.insert(MetadataKey::from_str(&key).unwrap(), value.parse().unwrap());
+    }
+
+    opentelemetry_otlp::new_pipeline()
+        .with_endpoint(endpoint.as_str())
+        .with_metadata(dbg!(metadata))
+        .with_tls_config(
+            ClientTlsConfig::new().domain_name(
+                endpoint
+                    .host_str()
+                    .expect("the specified endpoint should have a valid host"),
+            ),
+        )
+        .install()
+}
+
+const LEMONS_KEY: Key = Key::from_static_str("ex.com/lemons");
+const ANOTHER_KEY: Key = Key::from_static_str("ex.com/another");
+
+lazy_static::lazy_static! {
+    static ref COMMON_LABELS: [KeyValue; 4] = [
+        LEMONS_KEY.i64(10),
+        KeyValue::new("A", "1"),
+        KeyValue::new("B", "2"),
+        KeyValue::new("C", "3"),
+    ];
+}
+
+#[tokio::main]
+async fn main() -> Result<(), Box<dyn Error + Send + Sync + 'static>> {
+    let _guard = init_tracer()?;
```

Comment:
We usually use `_guard` to represent `Uninstall` structure instead of `(sdktrace::Tracer, opentelemetry_otlp::Uninstall)`. Give some clarity here.
```suggestion
    let (_, _guard) = init_tracer()?;
```